### ISSUE TITLE
False negative `SafeCast` with no braces

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#5478]
## Steps to Reproduce
```kotlin
    @Test
    fun `reports expression with no braces`() {
        val code = """
            fun test(element: Int) {
                val cast = if (element is Number) element else null
            }
        """.trimIndent()
        assertThat(subject.compileAndLint(code)).hasSize(1)
    }
```
